packages/flutter_tools/templates/plugin/lib/projectName_web.dart.tmpl

```diff
@@ -1,43 +1,25 @@
-import 'dart:async';
 // In order to *not* need this ignore, consider extracting the "web" version
 // of your plugin as a separate package, instead of inlining it in the same
 // package as the core of your plugin.
 // ignore: avoid_web_libraries_in_flutter
 import 'dart:html' as html show window;
 
-import 'package:flutter/services.dart';
 import 'package:flutter_web_plugins/flutter_web_plugins.dart';
 
-/// A web implementation of the {{pluginDartClass}} plugin.
-class {{pluginDartClass}}Web {
-  static void registerWith(Registrar registrar) {
-    final MethodChannel channel = MethodChannel(
-      '{{projectName}}',
-      const StandardMethodCodec(),
-      registrar,
-    );
+import '{{projectName}}_platform_interface.dart';
 
-    final pluginInstance = {{pluginDartClass}}Web();
-    channel.setMethodCallHandler(pluginInstance.handleMethodCall);
-  }
+/// A web implementation of the {{pluginDartClass}}Platform of the {{pluginDartClass}} plugin.
+class {{pluginDartClass}}Web extends {{pluginDartClass}}Platform {
+  /// Constructs a {{pluginDartClass}}Web
+  {{pluginDartClass}}Web();
 
-  /// Handles method calls over the MethodChannel of this plugin.
-  /// Note: Check the "federated" architecture for a new way of doing this:
-  /// https://flutter.dev/go/federated-plugins
-  Future<dynamic> handleMethodCall(MethodCall call) async {
-    switch (call.method) {
-      case 'getPlatformVersion':
-        return getPlatformVersion();
-      default:
-        throw PlatformException(
-          code: 'Unimplemented',
-          details: '{{projectName}} for web doesn\'t implement \'${call.method}\'',
-        );
-    }
+  static void registerWith(Registrar registrar) {
+    {{pluginDartClass}}Platform.instance = {{pluginDartClass}}Web();
   }
 
   /// Returns a [String] containing the version of the platform.
-  Future<String> getPlatformVersion() {
+  @override
+  Future<String?> getPlatformVersion() {
```

Comment:
@ditman Should we make this `async` so that it can just `return value;`, or is there a downside to that?